### ISSUE TITLE
Elevated, unconnected polys within same tile "break" findNearestPoly

### ISSUE THREAD

**Issue by TheSHEEEP** [recastnavigation/recastnavigation#417]
Another findNearestPoly problem, it seems.
But this one is unrelated to my previous issue: https://github.com/recastnavigation/recastnavigation/issues/416

First of all, here's an image showcasing the problem:

![4a0Tlzb](https://user-images.githubusercontent.com/1766205/71639347-a17bca00-2c7d-11ea-8a6b-bc1289c890c1.png)

The polygons on the elevated platform "connected" to the polygons on the floor with a red line are the problem.  
As you can see from their coloring, three of the polygons on the platform are in the same tile as those on the floor.  

This wouldn't be a problem in itself, but it leads to being unable to place agents or movement targets in the polygons on the platform (except for the one marked with a green line).
findNearestPoly doesn't find anything in these cases.

The other elevated parts of the navmesh (which you can also see in the image) work just fine. Especially the similar one in the distance (green circle) can have agents and movement targets placed upon it (which then can't really go anywhere, but that's as expected).

I assume the height is a problem in this case. If the platform was higher, it would have worked, but being not high enough, it seems like the polys on that non-working platform are placed within the same layer as the ones on the floor.  
Though this is just me guessing, might be an entirely different problem.

PS: I tried posting this on the discussion google group a while ago as I suspect this is just a matter of initialization/rasterization parameters or me not understanding something, and not really a bug, but it seems that didn't work, my post never appeared there.  
If it does eventually appear (and this is certain to not be a bug), feel free to just close this issue and answer on the group instead.

**Comment by memononen** [comment]
The coloring does not indicate that the polygons are "connected", they just
are stored in same tile/layer. Debug draw edge color indicates connectivity.

The case does not look special in any way.

While looking at your code earlier, the raycast magic you do when finding a
location on navmesh looks weird, there's no need for it. Have you tried
removing it? What are the values you pass to findNearestPoly(), where is
that location in the scene?

**Comment by TheSHEEEP** [comment]
> The coloring does not indicate that the polygons are "connected", they just are stored in same tile/layer. Debug draw edge color indicates connectivity.

But that's what I was saying?
They are obviously not connected, but in the same tile. And they are the only ones where those conditions are true and the only ones where findNearestPoly cannot find a ref.

> While looking at your code earlier, the raycast magic you do when finding a location on navmesh looks weird, there's no need for it. Have you tried removing it?

That code is taken from the recast demo itself.  
But it is indeed a remnant from the previous issue (I wasn't sure if the InputGeometry really was the same as the mesh in the game, so I did that to make sure we get a position on the InputGeometry which should be the closest it can get to the navmesh).
  
I removed it and sure enough that fixed the issue.
Which I don't really understand, to be honest... shouldn't the InputGeometry essentially be as close to the navmesh as possible?
It returns a position about 1.5 units lower than what the Godot raycast gave - and with that lower position cannot find anything in the navMesh, of course.  

And it only does that for this plateau, not anywhere else. Strange, but no need to solve that particular mystery.
All is good, for now ;)
